ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/BlockFetch/Client.hs

```diff
@@ -224,20 +296,29 @@ runBlockFetchTest BlockFetchClientTestSetup{..} = withRegistry \registry -> do
     LogicalClock.waitUntilDone clock
     atomically $ writeTVar varControlMessage Mux.Terminate
 
-    bfcoBlockFetchResults <- traverse waitThread blockFetchThreads
-    bfcoFetchedBlocks     <- readTVarIO varFetchedBlocks
-    bfcoTrace             <- getTrace
+    bfcoPeerOutcomes  <-
+      flip Map.traverseWithKey blockFetchThreads \peerId threadId -> do
+        blockFetchRes <- waitThread threadId
+        chainSyncRes  <- (Map.! peerId) <$> readTVarIO varChainSyncRes
```

Comment:
Hmm. Instead of this explicit mutvar stuff, could you also capture the ChainSync `ThreadId`, also send `Mux.Terminate` to the ChainSync client control vars, and read those threads' results the same as you're doing for BlockFetch? (Maybe you'd need to shutdown a peer's ChainSync before shutting down its BlockFetch, to appease `bracketSyncWithFetchClient`?)

Reply:
Related to https://github.com/input-output-hk/ouroboros-network/pull/3856#discussion_r1033527432